inc/deploygroup_dynamicdata.class.php

```diff
@@ -190,27 +190,66 @@ static function getDatas($itemtype, $params, array $forcedisplay=array()) {
     *
     * @since 0.85+1.0
     *
-    * @param object $group PluginFusioninventoryDeployGroup instance
-    * @return array
+    * @param group the group object
+    * @param use_cache retrieve computers_id from cache (computers_id_cache field)
+    * @return an array of computer ids
     */
-   static function getTargetsByGroup(PluginFusioninventoryDeployGroup $group) {
-      $search_params = PluginFusioninventoryDeployGroup::getSearchParamsAsAnArray($group, false,true);
-      if (isset($search_params['metacriteria']) && empty($search_params['metacriteria'])) {
-         unset($search_params['metacriteria']);
-      }
-      $search_params['sort'] = '';
+   static function getTargetsByGroup(PluginFusioninventoryDeployGroup $group, $use_cache = false) {
+      $ids = array();
+
+      if (!$use_cache || !$ids = self::retrieveCache($group)) {
```

Comment:
maybe the notation is a little too compact but here is the different cases:

- $use_cache = false, the first part of the 'if' will trigger (!$use_cache) and the second part will not (due to the '|' usage)
- $use_cache = true, the first part return false, so we try with the second part to load the cache:
  * it succeeds: we skip the block
  * it fails: we execute the block and load data from Search engine.

For resume, as the mentionned 'if' have an OR condition, if the first part is evaluated as true, the second part will not be evaluated.